docker-app/qfieldcloud/core/tests/test_secret.py

```diff
@@ -0,0 +1,183 @@
+                collaborator=cls.u3,
+                role=ProjectCollaborator.Roles.READER,
+            ),
+        ]
+
+        cls.p1.direct_collaborators.bulk_create(collaborators)
+
+    def _create_secret(self, **kwargs) -> Secret:
+        return Secret.objects.create(
+            type=Secret.Type.ENVVAR,
+            **kwargs,
+        )
+
+    def test_org_project_secret_available_for_user(self):
+        s1 = self._create_secret(name="s1", project=self.p1)
+
+        for user in [self.u1, self.u2, self.u3]:
+            with self.subTest():
+                secrets = Secret.objects.for_user_and_project(user, self.p1)
+
+                self.assertEqual(secrets.count(), 1)
+                self.assertEqual(secrets[0], s1)
+
+    def test_raises_when_user_is_not_org_project_collaborator(self):
+        _s1 = self._create_secret(name="s1", project=self.p1)
+
+        with self.assertRaises(UserProjectRoleError):
+            Secret.objects.for_user_and_project(self.u4, self.p1)
+
+    def test_raises_when_user_is_not_organization_member(self):
+        _s1 = self._create_secret(name="s1", project=self.p1)
+
+        with self.assertRaises(UserOrganizationRoleError):
+            Secret.objects.for_user_and_project(self.u5, self.p1)
+
+    def test_secret_not_available_for_random_org_project(self):
+        _s1 = self._create_secret(name="s1", project=self.p1)
+
+        secrets = Secret.objects.for_user_and_project(self.u1, self.p2)
+
+        self.assertEqual(secrets.count(), 0)
+
+    def test_assigned_secret_overrides_org_project_secret(self):
+        _project_secret = self._create_secret(name="s1", project=self.p1)
+        assigned_secret = self._create_secret(
+            name="s1", project=self.p1, assigned_to=self.u1
+        )
+
+        secrets = Secret.objects.for_user_and_project(self.u1, self.p1)
+
+        self.assertEqual(secrets.count(), 1)
+        self.assertEqual(secrets[0], assigned_secret)
+
+    def test_multiple_secrets_per_user_in_org_project(self):
+        _s1_project = self._create_secret(name="s1", project=self.p1)
+        s1_assigned = self._create_secret(
+            name="s1", project=self.p1, assigned_to=self.u1
+        )
+        s2_assigned = self._create_secret(
+            name="s2", project=self.p1, assigned_to=self.u1
+        )
+        s3_project = self._create_secret(name="s3", project=self.p1)
+        _s1_another_project = self._create_secret(
+            name="s1", project=self.p2, assigned_to=self.u1
+        )
+        _s1_another_user = self._create_secret(
+            name="s1", project=self.p1, assigned_to=self.u2
+        )
+
+        secrets = Secret.objects.for_user_and_project(self.u1, self.p1)
+
+        self.assertEqual(secrets.count(), 3)
+        self.assertEqual(secrets[0], s1_assigned)
+        self.assertEqual(secrets[1], s2_assigned)
+        self.assertEqual(secrets[2], s3_project)
+
+    def test_ind_project_secret_available_for_owner(self):
```

Comment:
```suggestion
    def test_ind_project_secret_available_for_owner(self):
```
what `ind` stands for? I know it was originally from me, but if you can read the cipher and rename properly...